File: RepoTransBench/target_projects/Java/C#/AllenDowney_ThinkJavaCode/tests/original/Ch01/HelloGoodbyeTest.cs
using Xunit;
using AllenDowney.ThinkJavaCode;
using System.IO;

namespace OriginalTests.Ch01
{
    public class HelloGoodbyeTest
    {
        [Fact]
        public void TestHelloOutput()
        {
            var originalOut = System.Console.Out;
            var baos = new StringWriter();
            System.Console.SetOut(baos);
            Hello.Main(System.Array.Empty<string>());
            System.Console.SetOut(originalOut);
            var output = baos.ToString().Trim();
            Assert.Equal("Hello, World!", output);
        }

        [Fact]
        public void TestGoodbyeOutput()
        {
            var originalOut = System.Console.Out;
            var baos = new StringWriter();
            System.Console.SetOut(baos);
            Goodbye.Main(System.Array.Empty<string>());
            System.Console.SetOut(originalOut);
            var output = baos.ToString().Replace("\r", "").Trim();
            Assert.Equal("Goodbye, cruel world", output);
        }
    }
}

File: RepoTransBench/source_projects/Java/AllenDowney_ThinkJavaCode/src/test/java/ch01/HelloGoodbyeTest.java
import org.junit.jupiter.api.Test;
import java.io.*;

import static org.junit.jupiter.api.Assertions.*;

public class HelloGoodbyeTest {

    @Test
    public void testHelloOutput() throws Exception {
        PrintStream originalOut = System.out;
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        System.setOut(new PrintStream(baos));
        Hello.main(new String[] {});
        System.setOut(originalOut);
        String output = baos.toString().trim();
        assertEquals("Hello, World!", output);
    }

    @Test
    public void testGoodbyeOutput() throws Exception {
        PrintStream originalOut = System.out;
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        System.setOut(new PrintStream(baos));
        Goodbye.main(new String[] {});
        System.setOut(originalOut);
        String output = baos.toString().replace("\r", "").trim();
        assertEquals("Goodbye, cruel world", output);
    }
}